crates/ruff_linter/src/checkers/ast/analyze/deferred_scopes.rs

```diff
@@ -89,6 +90,30 @@ pub(crate) fn deferred_scopes(checker: &mut Checker) {
             }
         }
 
+        if checker.enabled(Rule::RedefinedArgumentFromLocal) {
+            for (name, binding_id) in scope.bindings() {
+                for shadow in checker.semantic.shadowed_bindings(scope_id, binding_id) {
+                    let binding = &checker.semantic.bindings[shadow.binding_id()];
+                    if !matches!(
+                        binding.kind,
+                        BindingKind::LoopVar | BindingKind::BoundException
+                    ) {
+                        continue;
+                    }
+                    let shadowed = &checker.semantic.bindings[shadow.shadowed_id()];
+                    if !shadowed.kind.is_argument() {
+                        continue;
+                    }
```

Comment:
Is this supposed to check if the argument is unused? Or does it fire in either case?

Reply:
It's not checking the usage.

It looks sensible to check `binding.is_used()` as below
```rust
if !shadowed.kind.is_argument() || !binding.is_used() {
    continue;
}
```

In this case, results are following
```python
def foo(i):
    for i in range(10):    # emit error
        print(i)
        ...

def foo(i):
    for i in range(10):  # no error because it's unused
        ...
```
Is it good to go?

Reply:
I was referring to the inverse -- should this depend on the _argument_ being unused? I'll check Pylint.

Reply:
I erred on the side of removing the `!binding.is_used()` piece here. I think it still makes sense to raise, even if the new binding is unused, since it will cause the same problems?